app/models/admin_blacklisted_email.rb

```diff
@@ -1,12 +1,17 @@
 class AdminBlacklistedEmail < ApplicationRecord
   before_validation :canonicalize_email
+  after_create_commit :remove_invite_requests
```

Comment:
I think this should instead be `after_create` because removing the email from the invite requests is a db operation, so it should be part of the general create object db transaction. `after_commit` is more for things that aren't part of the db, like sending an email or updating elasticsearch, as they can't be rolled back if they are done with `after_create` (https://guides.rubyonrails.org/active_record_callbacks.html#after-commit-and-after-rollback)